docs/tutorial_email_generator.md

```diff
@@ -0,0 +1,266 @@
+>
+> Based on my previous experience, I have gained a deep understanding of the software development lifecycle and possess strong programming skills in languages such as Python, Java, and C++. I am well-versed in utilizing frameworks like TensorFlow and PyTorch, and have a solid foundation in data structures and algorithms. Additionally, I have successfully collaborated with cross-functional teams, demonstrating my ability to work effectively in fast-paced environments.
+>
+> I am highly motivated and eager to contribute my skills and knowledge to Google's mission of organizing the world's information and making it universally accessible and useful. I am ready to start as soon as possible and would be thrilled to be a part of such an innovative and impactful company.
+>
+> Thank you for considering my application. I have attached my resume for your review and would welcome the opportunity to discuss my qualifications further.
+>
+> Looking forward to hearing from you soon.
+>
+> Sincerely,  
+> [Your Name]  
+> [Contact Information]
+
+## Evaluating with LangCheck (No References)
+
+At this point in our development process, we would start iterating on the prompt template to try and improve the outputs. This is where LangCheck can come in and help - rather than having to manually prompt the app and check the outputs each time, we can automatically evaluate the outputs with the various LangCheck metrics!
+
+For example, in the Google application email output, we saw that the model added some details that we did not specify in the instructions. We can detect this by using the factual consistency metric, which checks whether the output text is consistent with the source text (in this case, we can set the `content_description` as the “source”). Indeed, we see that this output has a low factual consistency score.
+```python
+import langcheck
+
+factual_consistency = langcheck.metrics.factual_consistency(
+    generated_outputs=google_application['generated_email'],
+    sources=google_application['content_description'])
+print(f"factual consistency = {factual_consistency.metric_values[0]:.2f}")
+```
+
+```text
+factual consistency = 0.27
+```
+
+There are many factors other than factual consistency that we might care about in our email generator app. For example, we also want the email to be:
+- Not offensive/toxic
+- Fluent
+- Generally have a positive sentiment (though maybe not always)
+- Not contain any AI disclaimer language (i.e. language similar to “As an AI model, I don’t have opinions”)
+
+Here is how we can use LangCheck to check all of these aspects.
+```python
+emails = [item["generated_email"] for item in [japan_trip, running_late, google_application]]
+content_descriptions = [item["content_description"] for item in [japan_trip, running_late, google_application]]
```

Comment:
Nit: Can we format the code blocks with yapf so they're not super long?

Also, this is a style thing, but I feel like we don't need to specify the parameter names in our metric functions since they get super long. At least for the reference-free metrics with a single argument.

Reply:
> Nit: Can we format the code blocks with yapf so they're not super long?

Done!

> Also, this is a style thing, but I feel like we don't need to specify the parameter names in our metric functions since they get super long. At least for the reference-free metrics with a single argument.

Makes sense, I removed the param names for the metrics with just one argument. For the ones with two arguments, I kept the names since we have a mix of `sources=` and `reference_outputs=` in this tutorial, so I think it's helpful to make the differences clear

Reply:
LGTM!